packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -252,46 +249,78 @@ enum _LogType {
   trace
 }
 
-class _AnsiStatus extends Status {
-  _AnsiStatus(this.message, this.expectSlowOperation, this.onFinish) {
-    stopwatch = new Stopwatch()..start();
+class Spinner {
+  Spinner();
+  /// Use this factory to generate AnsiSpinner or Spinner as necessary, and
+  /// start them.
+  factory Spinner.forContextTerminal() {
+    if (terminal.supportsColor)
+      return new AnsiSpinner()..start();
+    return new Spinner()..start();
+  }
+  void start() {}
+  void stop() {}
+  void cancel() {}
+}
 
-    stdout.write('${message.padRight(52)}     ');
-    stdout.write('${_progress[0]}');
+/// Just a spinner, nothing more, nothing less.
+class AnsiSpinner extends Spinner {
+  int index = 0;
+  bool live = true;
+  Timer timer;
+
+  static final List<String> _progress = <String>['-', r'\', '|', r'/', '-', r'\', '|', '/'];
 
+  void _callback(Timer timer) {
+    stdout.write('\b${_progress[index]}');
+    index = ++index % _progress.length;
+  }
+
+  @override
+  void start() {
+    stdout.write(' ');
+    _callback(null);
     timer = new Timer.periodic(const Duration(milliseconds: 100), _callback);
   }
 
-  static final List<String> _progress = <String>['-', r'\', '|', r'/', '-', r'\', '|', '/'];
+  @override
+  void stop() {
+    if (!live)
+      return;
+    live = false;
+    timer.cancel();
+    stdout.write('\b \b');
```

Comment:
Yes, but because this program doesn't use the terminfo database and I actually drove a [heavily modified plutonium-powered DeLorean](https://en.wikipedia.org/wiki/DeLorean_time_machine) to work, I still think [things like this](http://ftp.wayne.edu/ldp/en/BackspaceDelete/BackspaceDelete-single.html#WHY) are an issue in the 21st century.  I'll drop it :-)